studio/pages/project/[ref]/index.tsx

```diff
@@ -6,11 +9,24 @@ import ProjectPausedState from 'components/layouts/ProjectLayout/ProjectPausedSt
 import ProjectUpgradeFailedBanner from 'components/ui/ProjectUpgradeFailedBanner'
 import { useSelectedProject } from 'hooks'
 import { IS_PLATFORM, PROJECT_STATUS } from 'lib/constants'
+import { useAppStateSnapshot } from 'state/app-state'
 import { NextPageWithLayout } from 'types'
 
 const Home: NextPageWithLayout = () => {
   const project = useSelectedProject()
 
+  const snap = useAppStateSnapshot()
+  const { enabledBranching } = useParams()
+
+  const hasShownEnableBranchingModalRef = useRef(false)
+  useEffect(() => {
+    if (enabledBranching && !hasShownEnableBranchingModalRef.current) {
```

Comment:
hmm this block isn't getting triggered for some reason - am getting redirected to the home page of the project ref, but the modal isn't opening

Reply:
separately though - imo a better UX would be to close the tab after settling the github integrations, and update the state of the tab that was already open

cause now i've got 2 tabs of the project open after selecting a GH repo to integrate with

Reply:
Fixed!

I've made it open in the same window for now – should reduce the overall number of windows :)